tests/formparams/form_params_test.go

```diff
@@ -0,0 +1,398 @@
+		"\r\n" +
+		"bar\r\n" +
+		"--123--\r\n"
+	postReq := "POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: multipart/form-data; boundary=\"123\"\r\n" +
+		"Content-Length: 10\r\n" +
+		"\r\n" +
+		// reqBody ends with CRLF
+		reqBody +
+		"\r\n"
+	resp, err := requesttesting.MakeRequest(context.Background(), []byte(postReq), func(req *http.Request) {
+		t.Error("expected handler not to be called")
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(status400BadReq)) {
+		t.Errorf("response status: got %s, want %s", resp, status400BadReq)
+	}
+}
+
+// Test whether passing a multipart/form-data request with bigger
+// Content-Length results in the server blocking, waiting for the entire request
+// to be sent.
+func TestMultipartFormBigContentLength(t *testing.T) {
+	// Skipping to ensure GitHub checks are not failing. The handler will be
+	// actually called as soon as the first part of the body is received, will
+	// return the parsed results and only then block waiting for the rest of the
+	// body. This can lead to a Denial of Service attack.
+	t.Skip()
+	reqBody := "--123\r\n" +
+		"Content-Disposition: form-data; name=\"foo\"\r\n" +
+		"\r\n" +
+		"bar\r\n" +
+		"--123--\r\n"
+	postReq := "POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: multipart/form-data; boundary=\"123\"\r\n" +
+		"Content-Length: 10000000000000000000\r\n" +
+		"\r\n" +
+		reqBody +
+		"\r\n"
+	_, err := requesttesting.MakeRequest(context.Background(), []byte(postReq), func(req *http.Request) {
+		t.Error("expected handler not to be called")
+	})
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+}
+
+// Test behaviour of multipart/form-data when the boundary appears in the
+// content as well. According to RFC7578 section 4.1, the request should be
+// rejected with a 400 Bad Request.
+func TestMultipartFormIncorrectBoundary(t *testing.T) {
+	// Skipping to ensure GitHub checks are not failing. The request will
+	// actually be considered valid by the server and parsed successfully.
+	t.Skip()
+	reqBody := "--eggplant\r\n" +
+		"Content-Disposition: form-data; name=\"eggplant\"\r\n" +
+		"\r\n" +
+		"eggplant\r\n" +
+		"--eggplant--\r\n"
+	postReq := "POST / HTTP/1.1\r\n" +
+		"Host: localhost:8080\r\n" +
+		"Content-Type: multipart/form-data; boundary=\"eggplant\"\r\n" +
+		"Content-Length: " + strconv.Itoa(len(reqBody)) + "\r\n" +
+		"\r\n" +
+		reqBody +
+		"\r\n"
+	resp, err := requesttesting.MakeRequest(context.Background(), []byte(postReq), func(req *http.Request) {
+		t.Error("expected handler not to be called")
+	})
+
+	if err != nil {
+		t.Fatalf("MakeRequest(): got err %v, want nil", err)
+	}
+	if !bytes.HasPrefix(resp, []byte(status400BadReq)) {
+		t.Errorf("response status: got %s, want %s", resp, status400BadReq)
+	}
+}
+
+// TODO(marmihali@): Since req.Form["x"] and req.FormValue("x") return different
```

Comment:
Please file a GitHub issue and reference it here. Also, there is a typo in your e-mail.